sgt-back-end/index.js

```diff
@@ -0,0 +1,194 @@
+var express = require('express');
+var app = express();
+
+app.use(express.json());
+
+const pg = require('pg');
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+app.get('/api/grades', (req, res, next) => {
+  const sql = `
+    select *
+      from "grades"
+  `;
+
+  db.query(sql)
+    .then(result => {
+        res.json(result.rows);
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occurred.'
+      });
+    });
+});
+
+app.post('/api/grades', (req, res, next) => {
+  if (req.body.name === undefined) {
+    res.status(400).json({
+      error: 'name is a required field'
+    });
+    return;
+  };
+
+  if (req.body.course === undefined) {
+    res.status(400).json({
+      error: 'course is a required field'
+    });
+    return;
+  };
+
+  if (req.body.score === undefined) {
+    res.status(400).json({
+      error: 'score is a required field'
+    });
+    return;
+  };
+
+  if ((req.body.score <= 0) || (req.body.score >= 101)) {
+    res.status(400).json({
+      error: 'score must be between 1 and 100'
+    });
+    return;
+  };
+
+  const name = req.body.name;
+  const course = req.body.course;
+  const score = req.body.score;
+  const params = [name, course, score]
+
+  const sql = `
+    insert into "grades" ("name", "course", "score")
+    values($1, $2, $3)
+    returning *
+  `;
+
+  db.query(sql, params)
+    .then(result => {
+      const grade = result.rows[0];
+      res.status(201);
+      res.json(grade);
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occurred.'
+      });
+    });
+});
+
+app.put('/api/grades/:gradeId', (req, res, next) => {
+  const gradeId = parseInt(req.params.gradeId, 10);
+  if (!Number.isInteger(gradeId) || gradeId <= 0) {
+    res.status(400).json({
+      error: '"gradeId" must be a positive integer'
+    });
+    return;
+  }
```

Comment:
Since `parseInt` forces the `gradeId` to be an integer if it's fractional (for example, `parseInt(13.54, 10)` returns `13`), so fractional `gradeId`s will pass, even though they shouldn't.